File: pset6/caesar/caesar.py
import cs50
import sys


def main():
    # Validate the key
    if len(sys.argv) > 1:
        # If the argv is not a digit then exit
        if sys.argv[1].isalpha() == True:
            print("Usage : python caesar.py k")
            exit(1)
        # Set k into argv[1] and set it to integer
        k = int(sys.argv[1])
        # Get the plaintext from the user
        p = cs50.get_string("plaintext: ")
        print("ciphertext: ", end="")
        # Loop i for the length of the plaintext
        for i in p:
            # If the current character is lowercase
            if i.islower():
                # Turn alphabet into ASCII code
                c = ord(i)
                # Turn ascii number into alphabet number ex: a = 1
                alphabet = int(c - 96)
                # Rotate alphabet after z back to a
                cipher = int(alphabet + k) % 26
                if cipher < c and cipher != 0:
                    # ((ASCII for lowercase a) - 1) + cipher
                    cipher = int(96 + cipher)
                    print(chr(cipher), end="")

            # If the current character is uppercase
            elif i.isupper():
                # Turn current character ASCII code
                c = ord(i)
                # Turn ASCII code into alphabet number ex: A = 1
                alphabet = int(c - 64)
                # Rotate alphabet from Z back to A
                cipher = int(alphabet + k) % 26
                if cipher < c and cipher != 0:
                    # ((ASCII for uppercase A) - 1) + cipher
                    cipher = int(64 + cipher)
                    print(chr(cipher), end="")
            else:
                print(i, end="")

        # Print new line after finished
        print("\n", end="")

    else:
        print("Usage : python caesar.py k")
        exit(1)


if __name__ == "__main__":
    main()

File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    // Validate the key
    if (argc == 2)
    {
        // Loop while the iteration times is less than length of key
        for (int n = 0; n < strlen(argv[1]); n++)
        {
            if (isdigit(argv[1][n]) == 0)
            {
                printf("Usage : ./caesar key\n");
                return 1;
            }
        }
        // Convert the string into int
        int k = atoi(argv[1]);
        // Get the plaintext from the user
        string p = get_string("plaintext: ");
        printf("ciphertext: ");
        // Loop while the iteration times is less than length of plaintext
        for (int i = 0; i < strlen(p); i++)
        {
            // Convert the char of plaintext into an int / an ascii
            int c = (int) p[i];
            // If the character is lowercase then do this
            if (islower(p[i]))
            {
                int alphabet = c - 96; // Turns ascii code into alphabet ex : a = 1
                int cipher = (alphabet + k) % 26; // Rotate alphabet after z back to a
                if (cipher < c && cipher != 0)
                {
                    cipher = 96 + cipher; // (Ascii lowercase a - 1) + cipher
                    printf("%c", cipher);
                }
            }
            // If the character is uppercase then do this
            else if (isupper(p[i]))
            {
                int alphabet = c - 64; // Turns ascii code into alphabet ex : A = 1
                int cipher = (alphabet + k) % 26; // Rotate alphabet from z back to a
                if (cipher < c && cipher != 0)
                {
                    cipher = 64 + cipher; // (Ascii of uppercase a - 1) + cipher
                    printf("%c", cipher);
                }
            }
            else
            {
                printf("%c", c);
            }
        }
        printf("\n");
    }
    else
    {
        printf("Usage : ./caesar key\n");
        return 1;
    }
}
